ui/ts/util/property.ts

```diff
@@ -0,0 +1,119 @@
+// source: util/property.ts
+// Author: Matt Tracy (matt@cockroachlabs.com)
+
+module Utils {
+  "use strict";
+
+  /**
+   * ReadOnlyProperty is a function that returns a value. 
+   *
+   * Each property also has an Epoch() function which returns a monotonically
+   * increasing number indicating the version of the property; if the value
+   * returned by the property changes, the epoch value will be increased.
+   */
+  export interface ReadOnlyProperty<T> {
+    (): T;
+    // Epoch returns the current update epoch for the property. This is a
+    // numeric value that monotonically increases each time the property is
+    // updated.
+    Epoch(): number;
```

Comment:
i don't know the ts type hierarchy, but this seems like a very generic type. could you tighten it to integer?